### PR TITLE
ci: Add system-tests

### PR DESCRIPTION
**Pull request by cbeauchesne**
Add a job in circleCI that run system-tests. It will prevent introducing bug tested in system tests.

For now, I only added the default scenario, up to C++ team to decide what to add or remove later.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -626,3 +658,7 @@ workflows:
         name: test << matrix.nginx-version >> on << matrix.base-image >>:<< matrix.arch >> WAF << matrix.waf >>
         requires:
         - build << matrix.nginx-version >> on << matrix.arch >> WAF << matrix.waf >>
+    - system_tests:
+        name: Run system tests
+        requires:
+        - build 1.25.4 on amd64 WAF ON
```
Comment (dmehala) [inline .circleci/config.yml:664]: This one is not necessary.

**Comment by dmehala** [comment]
LGTM. However I can see in the system-tests output:

> Library: cpp@6.6.6
> Weblog variant: nginx

How the weblog versioning is working? Is there a weblog versioning?

**Review by dmehala — APPROVED** [review]

**Comment by cbeauchesne** [comment]
aha, good catch, I didn't expect you to see this. There are several issue here.

Currently, the version number for c++ is the version number of this [present repo](https://github.com/DataDog/nginx-datadog/releases), where it should be the version of [dd-trace-cpp](https://github.com/DataDog/dd-trace-cpp/releases)

Second, I need to have a way to extract this version number from the `ngx_http_datadog_module.so` file, and I don't know how to do so. So shame on me, I went lazy, and I hardcoded a big version number, leaving this point for later.

If you know how to extract this version number, I can add it (it's inside system-tests repo). If you don't know, or if it's not possible (yet), my opinion is we can move forward with that hack, as long as the hardocded version is bigger than the current version, everything will works fine.

File: .circleci/config.yml (hunk 1 above)
Comment (cbeauchesne) [reply .circleci/config.yml:664]: 🤔 it's necessay for `build-and-test` but not for `build-and-test-all` ?

**Comment by cataphract** [comment]
> aha, good catch, I didn't expect you to see this. There are several issue here.
> 
> Currently, the version number for c++ is the version number of this [present repo](https://github.com/DataDog/nginx-datadog/releases), where it should be the version of [dd-trace-cpp](https://github.com/DataDog/dd-trace-cpp/releases)
> 
> Second, I need to have a way to extract this version number from the `ngx_http_datadog_module.so` file, and I don't know how to do so. So shame on me, I went lazy, and I hardcoded a big version number, leaving this point for later.
> 

They can be fetched like [this](https://github.com/DataDog/system-tests/pull/2455/files#diff-172cf498010a2e06a0e2e094374c274c96584c1eb6400c729a70be25ed0b17acR7).


See also: https://github.com/DataDog/nginx-datadog/blob/master/src/version.cpp.in

**Comment by cbeauchesne** [comment]
thanks @cataphract , let me add this

**Comment by dmehala** [comment]
> They can be fetched like [this](https://github.com/DataDog/system-tests/pull/2455/files#diff-172cf498010a2e06a0e2e094374c274c96584c1eb6400c729a70be25ed0b17acR7).

That's smart. We should also those this for `dd-trace-cpp` instead of parsing `version.h`. This would have avoid the issue with prod we got this week

**Comment by cbeauchesne** [comment]
@dmehala could you merge the Pr? I don't have the right to do it myself 🙏

**Pull request merged**